File: java/BillsApi/src/main/java/com/apilangarena/javabillsapi/infrastructure/persistence/entity/BillLineEntity.java
package com.apilangarena.javabillsapi.infrastructure.persistence.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

import java.math.BigDecimal;

@Entity
@Table(name = "bill_line")
public class BillLineEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "bill_id", nullable = false)
    private BillEntity bill;

    @Column(name = "line_no", nullable = false)
    private Integer lineNo;

    @Column(name = "concept", nullable = false, length = 200)
    private String concept;

    @Column(name = "quantity", nullable = false, precision = 10, scale = 2)
    private BigDecimal quantity;

    @Column(name = "unit_amount", nullable = false, precision = 12, scale = 2)
    private BigDecimal unitAmount;

    @Column(name = "line_amount", nullable = false, precision = 12, scale = 2)
    private BigDecimal lineAmount;

    public BillLineEntity() {
    }

    public BillLineEntity(Integer lineNo, String concept, BigDecimal quantity, BigDecimal unitAmount, BigDecimal lineAmount) {
        this.lineNo = lineNo;
        this.concept = concept;
        this.quantity = quantity;
        this.unitAmount = unitAmount;
        this.lineAmount = lineAmount;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public BillEntity getBill() {
        return bill;
    }

    public void setBill(BillEntity bill) {
        this.bill = bill;
    }

    public Integer getLineNo() {
        return lineNo;
    }

    public void setLineNo(Integer lineNo) {
        this.lineNo = lineNo;
    }

    public String getConcept() {
        return concept;
    }

    public void setConcept(String concept) {
        this.concept = concept;
    }

    public BigDecimal getQuantity() {
        return quantity;
    }

    public void setQuantity(BigDecimal quantity) {
        this.quantity = quantity;
    }

    public BigDecimal getUnitAmount() {
        return unitAmount;
    }

    public void setUnitAmount(BigDecimal unitAmount) {
        this.unitAmount = unitAmount;
    }

    public BigDecimal getLineAmount() {
        return lineAmount;
    }

    public void setLineAmount(BigDecimal lineAmount) {
        this.lineAmount = lineAmount;
    }
}

File: kotlin/BillsApi/src/main/kotlin/com/apilangarena/billsapi/infrastructure/persistence/entity/BillLineEntity.kt
package com.apilangarena.billsapi.infrastructure.persistence.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table
import java.math.BigDecimal

@Entity
@Table(name = "bill_line")
class BillLineEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    var id: Long? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "bill_id", nullable = false)
    var bill: BillEntity? = null,

    @Column(name = "line_no", nullable = false)
    var lineNo: Int = 0,

    @Column(name = "concept", nullable = false, length = 200)
    var concept: String = "",

    @Column(name = "quantity", nullable = false, precision = 10, scale = 2)
    var quantity: BigDecimal = BigDecimal.ZERO,

    @Column(name = "unit_amount", nullable = false, precision = 12, scale = 2)
    var unitAmount: BigDecimal = BigDecimal.ZERO,

    @Column(name = "line_amount", nullable = false, precision = 12, scale = 2)
    var lineAmount: BigDecimal = BigDecimal.ZERO,
)
